pkg/restore/restore.go

```diff
@@ -578,6 +590,97 @@ func addToResult(r *api.RestoreResult, ns string, e error) {
 	}
 }
 
+func (ctx *context) waitForPVDeletion(name string, resourceClient client.Dynamic) (bool, error) {
+	pvLogger := ctx.log.WithField("pvName", name)
+
+	var shouldRestore bool
+	err := wait.PollImmediate(time.Second, ctx.resourceTerminatingTimeout, func() (bool, error) {
+		clusterPV, err := resourceClient.Get(name, metav1.GetOptions{})
+
+		if apierrors.IsNotFound(err) {
+			pvLogger.Debug("PV not found, safe to restore")
+			// PV not found, can safely exit loop and proceed with restore.
+			shouldRestore = true
+			return true, nil
+		}
+
+		if err != nil {
+			return true, errors.Wrapf(err, "could not retrieve in-cluster copy of PV %s", name)
```

Comment:
doesn't really matter functionally, since the `PollImmediate` will break if it gets either a `true` or an error, but I'd probably return `false, errors.Wrapf(...)` here and in the subsequent error cases

Reply:
What's the reasoning for using `false`? My reasoning for returning `true` was to be consistent in the cases where the loop was being broken.

Reply:
Semantically, the first return argument indicates if the condition that's being checked for has been met, where the condition here is something like "we've determined whether the PV should be restored". IMHO, in an error case, we're breaking not because we've determined that, but because we've hit an error.

`false` makes more sense to me but since functionally it's the same either way, if you prefer it as-is then I'm fine leaving it.

Reply:
Ok, that explanation makes sense to me. I'll switch it.